src/wp-includes/blocks.php

```diff
@@ -1254,7 +1254,31 @@ function wp_enqueue_block_style( $block_name, $args ) {
 
 	$hook = did_action( 'wp_enqueue_scripts' ) ? 'wp_footer' : 'wp_enqueue_scripts';
 	if ( wp_should_load_separate_core_block_assets() ) {
-		$hook = "render_block_$block_name";
+		/**
+		 * Callback function to register and enqueue styles.
+		 *
+		 * @param string $content The block content.
+		 * @param array  $block   The full block, including name and attributes.
+		 * @return string Block content.
+		 */
+		$callback_separate = static function( $content, $block ) use ( $block_name, $callback ) {
```

Comment:
Anonymous functions as callbacks to filter/action is prohibited in Core.

To resolve, an overall architectural strategy is needed to provide the means to unhook callbacks and remove the anonymous functions.

However, as this pattern already exists in this file and it's too late in 5.9 to introduce a new architectural strategy to remove the need for these lambdas, this pattern is consistent. Plus the `static` keyword is for performance.